src/ja/server/database/types/work_machine.py

```diff
@@ -93,29 +119,41 @@ def __init__(self,
         @param resources The available resources on the work machine.
         @param connection The connection information about the work machine.
         """
+        self._uid = uid
+        self._state = state
+        self._resources = resources
+        self._connection = connection
 
     @property
     def uid(self) -> str:
         """!
         @return The UID of the work machine.
         """
+        return self._uid
 
     @property
     def state(self) -> WorkMachineState:
         """!
         @return: Whether the work machine is online, retired, or offline.
         """
+        return self._state
 
     @property
     def resources(self) -> WorkMachineResources:
         """!
         @return The resources of the work machine, if it is online,
           otherwise None.
         """
+        if self._state == WorkMachineState.ONLINE:
```

Comment:
A retired work machine can also still have resources.
Retired means that the work machine will not receive new jobs and is pending shutdown but existing jobs will continue to run until they are done.